lib/rouge/lexers/markdown.rb

```diff
@@ -130,53 +130,57 @@ def html
         rule %r/\n/, Text
       end
 
+      state :ref do
+        rule %r/\s+(?=["'(])/ do
+          token Text
+          goto :title
+        end
+
+        rule(//) { pop! }
+      end
+
       state :link do
         rule %r/(\[)(#{edot}*?)(\])/ do
           groups Punctuation, Str::Symbol, Punctuation
           pop!
         end
 
-        rule %r/[(]/ do
-          token Punctuation
-          push :inline_title
-          push :inline_url
-        end
+        rule %r/[(]/, Punctuation, :url
+        rule %r/[)]/, Punctuation, :pop!
 
-        rule %r/[ \t]+/, Text
+        rule %r/[ \t]+/, Text, :title
```

Comment:
This was pre-existing, but I wonder why this used `_\t` here instead of `\s` which would cover all whitespace (underscore used to emphasize literal space).